x-pack/plugins/enterprise_search/public/applications/app_search/index.tsx

```diff
@@ -69,11 +72,17 @@ export const AppSearchConfigured: React.FC<IInitialAppData> = (props) => {
           ) : (
             <Switch>
               <Route exact path={ROOT_PATH}>
-                <Redirect to={ENGINES_PATH} />
-              </Route>
-              <Route exact path={ENGINES_PATH}>
-                <EngineOverview />
+                {canViewEngines ? (
+                  <Redirect to={ENGINES_PATH} />
+                ) : (
+                  <NotFound product={APP_SEARCH_PLUGIN} />
+                )}
               </Route>
+              {canViewEngines && (
```

Comment:
This one's a little weird, would be interested in all y'alls thoughts here or if you think it's OK 🤔 

Historically, I think the reason why we had the `canViewEngines` check was for the (now defunct) Finance role, which didn't have access to engines but did have access to billing settings.

That no longer applies in a SMAS world, so in theory every role/user _should_ have access to engines... but I left the conditional anyway in just in case, with a fallback to our 404 view* 🤷 

If we think it's unnecessary though, I can definitely get rid of the `canViewEngines` wrappers and just leave the EngineOverview always visible. Yay/nay/just leave it as-is?

> \* For even more context - [in ent-search, the fallback is to the Settings page](https://github.com/elastic/ent-search/blob/master/app/javascript/app_search/App/App.tsx#L100-L103), but that also no longer applies in a SMAS world. In SaaS, I believe every user had access to Settings so they could change their password, but that no longer applies in SMAS/native auth, where we don't manage passwords & only owners/admins have access to account settings.

Reply:
+1 for pulling it out.  Is there somewhere all the rules and abilities for Kibana users are documented outside of code?

Reply:
Good question, I think this is something we actually might want to bring product in on. I'll start a convo w/ Casey & Nick!

Reply:
Just to close the loop on this: Casey is good with removing the `canViewEngines` check 🎉 

Re: your question Byron:

> Is there somewhere all the rules and abilities for Kibana users are documented outside of code?

He linked https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-users.html - does that apply/answer your question? 🤔

Reply:
I removed `canViewEngines` entirely in this PR: https://github.com/elastic/kibana/pull/77693/commits/7d1d46970d705bf941d053857257dd77c8d4864e

But now I'm actually not sure I like it hahaha. It kind of leaves a weird hole (in my heart?). I'm probably overthinking it. Thumbs up, thumbs down?